### PR TITLE
docs: add helper commands to contributing.md

### PR DESCRIPTION
**Pull request by azro352**
## Changes

Add some helper commands for local testing, to help contributers

## Context

#25459

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: .github/contributing.md [hunk 1]
```diff
@@ -31,8 +31,30 @@ Before you start any Pull Request, it's recommended that you open a [discussion]
 That way you can be sure that the maintainer(s) agree on what to change and how, and you can hopefully get a quick merge afterwards.
 Also, let the maintainers know that you plan to work on a particular issue so that no one else starts any duplicate work.
 
+### Tests
+
 Pull Requests can only be merged once all status checks are green, which means `pnpm test` passes, and coverage is 100%.
 
+To help you execute your tests, here are some helper commands:
```
Comment (HonkingGoose) [inline .github/contributing.md:38]:
```suggestion
Use these commands to help run your tests:
```

File: .github/contributing.md [hunk 2]
```diff
@@ -31,8 +31,30 @@ Before you start any Pull Request, it's recommended that you open a [discussion]
...
+
+- to execute a single test file
```
Comment (HonkingGoose) [inline .github/contributing.md:40]:
```suggestion
- To run a single test file
```

File: .github/contributing.md [hunk 3]
```diff
@@ -31,8 +31,30 @@ Before you start any Pull Request, it's recommended that you open a [discussion]
...
+  ```bash
+  pnpm run jest ./lib/modules/platform/gitlab/index.spec.ts
+  ```
+- to execute a single test batch, the `-t` value should be part of the `describe` value of the test batch
```
Comment (HonkingGoose) [inline .github/contributing.md:44]:
```suggestion
- To run a single test batch, the `-t` value must be part of the `describe` value of the test batch
```

File: .github/contributing.md [hunk 4]
```diff
@@ -31,8 +31,30 @@ Before you start any Pull Request, it's recommended that you open a [discussion]
...
+  ```bash
+  pnpm run jest ./lib/modules/platform/gitlab/index.spec.ts -t "getJsonFile"
+  ```
+- to execute a single test, the `-t` value should be part of the `it` value of the test batch
```
Comment (HonkingGoose) [inline .github/contributing.md:48]:
```suggestion
- To run a single test, the `-t` value must be part of the `it` value of the test batch
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: .github/contributing.md [hunk 5]
```diff
@@ -31,8 +31,30 @@ Before you start any Pull Request, it's recommended that you open a [discussion]
...
 
+### Tests
+
 Pull Requests can only be merged once all status checks are green, which means `pnpm test` passes, and coverage is 100%.
 
+Use these commands to help run your tests:
+
+- To run a single test file
+  ```bash
+  pnpm run jest ./lib/modules/platform/gitlab/index.spec.ts
```
Comment (rarkins) [inline .github/contributing.md:42]:
```suggestion
  pnpm jest platform/gitlab/index
```

File: .github/contributing.md [hunk 6]
```diff
@@ -31,8 +31,30 @@ Before you start any Pull Request, it's recommended that you open a [discussion]
...
+  ```
+- To run a single test batch, the `-t` value must be part of the `describe` value of the test batch
+  ```bash
+  pnpm run jest ./lib/modules/platform/gitlab/index.spec.ts -t "getJsonFile"
```
Comment (rarkins) [inline .github/contributing.md:46]:
```suggestion
  pnpm run jest platform/gitlab/index -t "getJsonFile"
```

File: .github/contributing.md [hunk 7]
```diff
@@ -31,8 +31,30 @@ Before you start any Pull Request, it's recommended that you open a [discussion]
...
+  ```
+- To run a single test, the `-t` value must be part of the `it` value of the test batch
+  ```bash
+  pnpm run jest ./lib/modules/platform/gitlab/index.spec.ts -t "returns file content from given repo"
```
Comment (rarkins) [inline .github/contributing.md:50]:
```suggestion
  pnpm jest platform/gitlab/index -t "returns file content from given repo"
```

**Review by rarkins — CHANGES_REQUESTED** [review]
Simplified command

File: .github/contributing.md [hunk 8]
```diff
@@ -31,8 +31,30 @@ Before you start any Pull Request, it's recommended that you open a [discussion]
 That way you can be sure that the maintainer(s) agree on what to change and how, and you can hopefully get a quick merge afterwards.
 Also, let the maintainers know that you plan to work on a particular issue so that no one else starts any duplicate work.
 
+### Tests
+
 Pull Requests can only be merged once all status checks are green, which means `pnpm test` passes, and coverage is 100%.
 
+Use these commands to help run your tests:
+
+- To run a single test file
```
Comment (rarkins) [inline .github/contributing.md:40]:
```suggestion
- To run a single test folder
  ```bash
  pnpm jest platform/gitlab
  ```
- To run a single test file
```

File: .github/contributing.md [hunk 9]
```diff
@@ -31,8 +31,32 @@ Before you start any Pull Request, it's recommended that you open a [discussion]
...
 Pull Requests can only be merged once all status checks are green, which means `pnpm test` passes, and coverage is 100%.
 
+Use these commands to help run your tests:
+
+- To run a single test folder
+  ```bash
+  pnpm jest platform/gitlab
+  ```bash
+  pnpm jest platform/gitlab/index
+  ```
```
Comment (rarkins) [inline .github/contributing.md:45]:
```suggestion
- To run a single test folder
  ```bash
  pnpm jest platform/gitlab
  ```
- to run against a single test file
  ```bash
  pnpm jest platform/gitlab/index
  ```
```

File: .github/contributing.md [hunk 10]
```diff
@@ -31,8 +31,41 @@ Before you start any Pull Request, it's recommended that you open a [discussion]
...
+- To run against a single test file, specify down to the filename (suffix is not necessary)
+
+  ```bash
+  pnpm jest platform/gitlab/index
+  ```
+  
+- To run a single test batch, the `-t` value must be part of the `describe` value of the test batch
+
+  ```bash
+  pnpm run jest platform/gitlab/index -t "getJsonFile"
```
Comment (azro352) [inline .github/contributing.md:55]:
```suggestion
  pnpm jest platform/gitlab/index -t "getJsonFile"
```

**Review by rarkins — DISMISSED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**